File: Recursion/Code/powerOf2.cpp
//Program to calculate 2 raised to the power n

#include <iostream>

using namespace std;

int power(int n){
    if (n==1){                 //Base Case
        return 2;
    }
    else{
        return 2*power(n-1);   //multiplication with 2 is small calculation & power(n-1) is the recursive call
    }
}

int main() {
    int n;
    cout<< "Enter a positive number : ";
    cin>>n;
    cout<<"2^"<<n<<" = "<<power(n)<<endl;
    return 0;
}

File: Recursion/Code/powerOf2.java
import java.util.*;
class power
 {

    static int power(int n)
    {
		if(n==1)  // base condition;
		return 2;

		int recResult = power(n-1);
		int smallCal = 2*recResult;  

		return smallCal;
    }

    // Driver Code
    public static void main(String args[])
    {
		Scanner in = new Scanner(System.in);
		System.out.print("Enter a positive number : ");
		int n = in.nextInt();

        System.out.println("Factorial of " + n + " is " + power(n));
    }
}
